cng/hash_test.go

```diff
@@ -181,6 +181,61 @@ func TestHash_OneShot(t *testing.T) {
 	}
 }
 
+func TestHashAllocations(t *testing.T) {
+	msg := []byte("testing")
+	n := int(testing.AllocsPerRun(10, func() {
+		sink ^= cng.MD4(msg)[0]
+		sink ^= cng.MD5(msg)[0]
+		sink ^= cng.SHA1(msg)[0]
+		sink ^= cng.SHA256(msg)[0]
+		sink ^= cng.SHA384(msg)[0]
+		sink ^= cng.SHA512(msg)[0]
```

Comment:
Does it help in some way to put all of these in the same test?

Or could these be separate tests (perhaps using `t.Run`) to make it easier to figure out what went wrong when something changes?

Reply:
Good recommendation. We have so many tests with this approach so I actually used the same code for most of the tests, I'll see if we can cut any lines or make it look simpler with that approach on the other repositories too.